src/api/topics/topics.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, UpdateResult } from 'typeorm';
+
+import { CreateTopicDto } from '@api/topics/dto/create-topic.dto';
+import { UpdateTopicDto } from '@api/topics/dto/update-topic.dto';
+import { TopicEntity } from '@api/topics/topic.entity';
+
+@Injectable()
+export class TopicsService {
+  constructor(
+    @InjectRepository(TopicEntity)
+    private topicRepository: Repository<TopicEntity>,
+  ) {}
+
+  create(createTopicDto: CreateTopicDto): Promise<TopicEntity> {
+    return this.topicRepository.save(createTopicDto);
+  }
+
+  findAll(): Promise<TopicEntity[]> {
+    return this.topicRepository.find();
+  }
+
+  findOne(id: number): Promise<TopicEntity | null> {
+    return this.topicRepository.findOne({
+      where: { id },
+    });
+  }
+
+  update(id: number, updateTopicDto: UpdateTopicDto): Promise<UpdateResult> {
+    const topic = this.findOne(id);
```

Comment:
The method findOne returns a promise, so here you should await it

Reply:
😅 yup, I think I accidentally deleted it